partner_bank_code/models/res_bank.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 Ecosoft Co., Ltd. (https://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class ResBank(models.Model):
+    _inherit = "res.bank"
+
+    bank_code = fields.Char()
+    bank_branch_code = fields.Char()
+
+    _sql_constraints = [
+        (
+            "bank_code_unique",
+            "unique(bank_code, bank_branch_code)",
+            "Bank and Branch Code should be unique.",
+        ),
+    ]
+
+    def name_get(self):
+        """Add bank and branch code to name if available"""
+        result = super().name_get()
+        return [
+            (
+                _id,
+                name
+                + (
+                    (
+                        " [%s%s]"
+                        % (
+                            self.browse(_id).bank_code,
+                            ("/%s" % self.browse(_id).bank_branch_code)
+                            if self.browse(_id).bank_branch_code
+                            else "",
+                        )
+                    )
+                    if self.browse(_id).bank_code
+                    else ""
+                ),
+            )
+            for _id, name in result
+        ]
```

Comment:
I just wondered whether the repeated browse could be prevented with a little helper function, something along the lines off:
```
def make_name(id, name):
    bank = self.browse(id)
    return " ".join(
        [
            name,
            "/".join(
                [
                    part
                    for part in [bank.bank_code, bank.bank_branch_code]
                    if part
                ]
            )
        ]
    )
result = super().name_get()
return [(id, make_name(id, name)) for id, name in result]
```

Reply:
we could, but nowadays this just looks at the same cache, so that's basically syntactic sugar

Reply:
It was  a suggestion. I might be old fashioned in not wanting to retrieve the same data repeatedly, even though I am aware of the caching. But I approved the PR anyway already.